src/frontends/tensorflow/src/translate_session.cpp

```diff
@@ -154,6 +159,124 @@ size_t get_flat_index_by_name_and_id(const ov::frontend::NamedOutputVector& outp
         return idx;
     }
 }
+
+// create Parameter node that will produce given tensor
+std::shared_ptr<ov::opset8::Parameter> create_parameter_node_for_tensor(ov::Output<ov::Node> output_tensor) {
+    auto param =
+        std::make_shared<ov::opset8::Parameter>(output_tensor.get_element_type(), output_tensor.get_partial_shape());
+    param->output(0).set_names(output_tensor.get_names());
+    output_tensor.replace(param->output(0));
+    return param;
+}
+
+void fuse_loop_cond(std::shared_ptr<LoopCond>& loop_cond, OpMap& ov_tensors_map) {
+    // ov_tensors_map maps a operation name to a vector of its output tensors
+    auto node_name = loop_cond->get_friendly_name();
+    // find key points for condition and body graphs
+    FRONT_END_GENERAL_CHECK(loop_cond, "[TensorFlow Frontend] internal error: pointer to LoopCond node is nullptr");
+
+    // extract condition and body graphs
+    // scan LoopCond node vicinity
+    // 1. LoopCond has just one output
+    // walk through all consuming inputs that are expected to be only for Switch nodes
+    std::vector<std::shared_ptr<Switch>> switch_nodes;
+    for (const auto& consuming_input : loop_cond->get_output_target_inputs(0)) {
+        auto switch_node = ov::as_type_ptr<Switch>(consuming_input.get_node()->shared_from_this());
+        FRONT_END_GENERAL_CHECK(switch_node,
+                                "[TensorFlow Frontend] internal error or inconsistent model: consumer of LoopCond "
+                                "output is not Switch operation");
+        switch_nodes.push_back(switch_node);
+    }
+
+    // collect all output tensors for Loop
+    // the created Loop node outputs will be connected with ov_outputs
+    size_t num_inputs = switch_nodes.size();
```

Comment:
Do we need to check it is greater than 0?

Reply:
No, that is because `LoopCond` cannot exist without `Switch` consuming node(s)